.github/workflows/build-plugin-archive.yml

```
@@ -0,0 +1,241 @@
+
+      - name: Run Composer Assets Compiler
+        if: steps.composer-tools.outputs.assets-compiler == '0'
+        run: composer compile-assets
+
+      - name: Run WP Translation Downloader
+        if: steps.composer-tools.outputs.translation-downloader == '0'
+        run: composer wp-translation-downloader:download
+
+      - name: Install Composer dependencies without dev dependencies
+        # ⬑ Now we want to get rid of dev dependencies and most probably skip composer scripts
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+
+      #
+      # JS environment. Only for setting up node/npm versions. Actual tasks should be run at project level
+      #
+
+      - name: Check for package.json
+        id: check-for-package
+        run: |
+          if [ -f package.json ]; then
+            echo "has_package=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_package=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up node
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+
+      - name: Cache npm dependencies
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      #
+      # Package metadata
+      #
+
+      - name: Add commit hash to plugin header
+        env:
+          GIT_SHA: ${{ steps.git.outputs.sha_short }}
+        run: 'sed -Ei "s/SHA: .*/SHA: ${GIT_SHA}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      - name: Set plugin version header
+        env:
+          PACKAGE_VERSION: ${{ github.event.inputs.packageVersion }}
+        run: 'sed -Ei "s/Version: .*/Version: ${PACKAGE_VERSION}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      #
+      # Archive tooling
+      #
+
+      - name: Install WP-CLI
+        run: |
+          mkdir ~/wp-cli && cd ~/wp-cli
+          curl -L https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar --output wp
+          chmod +x wp
+          sudo ln -s ~/wp-cli/wp /usr/local/bin/wp
+
+      - name: Install dist-archive command
+        run: wp package install wp-cli/dist-archive-command
+
+      - name: Create dist-archive php parameters from inputs
+        id: dist-archive
+        run: |
+          PARAMS=""
+          FILE='${{ inputs.EXTRA_PHP_FILE }}'
+          CODE=$(cat <<'EOF'
+          ${{ inputs.EXTRA_PHP }}
+          EOF
+          )
+          if [ ! -z "$FILE" ]; then
+            PARAMS+="--require=$FILE "
+          fi
+          if [ ! -z "$CODE" ]; then
+            PARAMS+="--exec=${CODE@Q} "
+          fi
+          echo "params=${PARAMS}" >> $GITHUB_OUTPUT
```

TBH, I don't understand why we can't utilize GitHub Actions for running PHP commands and need to use the overly complex setup to build `--require`/`--exec` flags for WP-CLI.
```suggestion
      - name: Execute custom code before archive creation
        shell: php {0}
        run: |
          ${{ inputs.EXTRA_PHP_FILE }}
```
(untested!)
Documentation: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#custom-shell
Example: https://github.com/marketplace/actions/setup-php-action#inline-php-scripts

---

The difference is that this executes naked PHP without any context whatsoever.
If passed to WP-CLI instead, you execute within its [bootstrapping process](https://github.com/wp-cli/wp-cli/blob/2efbaad2d5def76c5fb78b758f98f3b9c734e694/php/bootstrap.php#L28-L29), giving you access to its autoloader, shell utils and possibly also limited access to its execution context.

Granted: It unfortunately does not provide you with a bootstrapped WordPress (->access to core functions), but I'd say this is still miles better than _raw PHP_  (at that point, I'd prefer bash/sh for its vastly more powerful file/directory modification tools).